src/main/java/lotto/domain/LottoBalls.java

```diff
@@ -0,0 +1,52 @@
+package lotto.domain;
+
+import lotto.message.ErrorMessage;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+
+public class LottoBalls {
+    public static final int LOTTO_BALL_VALID_COUNT = 6;
+    private final Set<LottoNumber> balls = new TreeSet<>();
+
+    public LottoBalls(Collection<Integer> balls) {
+        this.balls.addAll(balls.stream().map(LottoNumber::new).collect(Collectors.toSet()));
+        validate();
+    }
+
+    public LottoBalls(Integer... balls) {
+        this(Arrays.asList(balls));
+    }
+
+    private void validate() {
+        if(balls.size() != LOTTO_BALL_VALID_COUNT){
+            throw new RuntimeException(String.format(ErrorMessage.INVALID_BALL_COUNT, LOTTO_BALL_VALID_COUNT));
+        }
+    }
+
+    @Override
+    public String toString() {
+        return balls.stream()
+                .map(Object::toString)
+                .collect(Collectors.joining(", "));
+    }
+
+    public int containsCount(LottoBalls lottoBalls) {
+        return (int) balls.stream().filter(lottoBalls.balls::contains).count();
+    }
```

Comment:
```suggestion
    public int containsCount(LottoBalls lottoBalls) {
        return (int) balls.stream().filter(lottoBalls::contains).count();
    }
```
밑에서 contatins 메소드를 구현했기 때문에 이렇게 표현할 수 있습니다.